provider/provider.go

```diff
@@ -0,0 +1,182 @@
+// Copyright 2022 CloudWeGo Authors.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package provider
+
+import (
+	"context"
+
+	"github.com/cloudwego/kitex/pkg/klog"
+	runtimemetrics "go.opentelemetry.io/contrib/instrumentation/runtime"
+	"go.opentelemetry.io/otel"
+	"go.opentelemetry.io/otel/exporters/otlp/otlpmetric"
+	"go.opentelemetry.io/otel/exporters/otlp/otlpmetric/otlpmetricgrpc"
+	"go.opentelemetry.io/otel/exporters/otlp/otlptrace"
+	"go.opentelemetry.io/otel/exporters/otlp/otlptrace/otlptracegrpc"
+	"go.opentelemetry.io/otel/metric/global"
+	controller "go.opentelemetry.io/otel/sdk/metric/controller/basic"
+	"go.opentelemetry.io/otel/sdk/metric/export/aggregation"
+	processor "go.opentelemetry.io/otel/sdk/metric/processor/basic"
+	"go.opentelemetry.io/otel/sdk/metric/selector/simple"
+	"go.opentelemetry.io/otel/sdk/resource"
+	sdktrace "go.opentelemetry.io/otel/sdk/trace"
+	semconv "go.opentelemetry.io/otel/semconv/v1.7.0"
+)
+
+type OtelProvider interface {
+	Shutdown(ctx context.Context) error
+}
+
+type otelProvider struct {
+	traceExp      *otlptrace.Exporter
+	metricsPusher *controller.Controller
+}
+
+func (p *otelProvider) Shutdown(ctx context.Context) error {
+	var err error
+
+	if p.traceExp != nil {
+		if err = p.traceExp.Shutdown(ctx); err != nil {
+			otel.Handle(err)
+		}
+	}
+
+	if p.metricsPusher != nil {
+		if err = p.metricsPusher.Stop(ctx); err != nil {
+			otel.Handle(err)
+		}
+	}
+
+	return err
+}
+
+// NewOpenTelemetryProvider Initializes an otlp trace and metrics provider
+func NewOpenTelemetryProvider(opts ...Option) *otelProvider {
+	var (
+		err           error
+		traceExp      *otlptrace.Exporter
+		metricsPusher *controller.Controller
+	)
+
+	ctx := context.TODO()
+
+	cfg := newConfig(opts)
+
+	if !cfg.enableTracing && cfg.enableMetrics {
```

Comment:
it should be `if !cfg.enableTracing && !cfg.enableMetrics ` ?

Reply:
> it should be `if !cfg.enableTracing && !cfg.enableMetrics ` ?

yes my fault